File: utils/safety_calculations.py
"""Illustrative safety risk scoring and summaries."""

from __future__ import annotations
import pandas as pd


def safety_kpis(events: pd.DataFrame) -> dict[str, str | int]:
    open_events = events[events["status"] != "Closed"]
    return {
        "Open safety events": len(open_events),
        "Critical events": int(((open_events["severity"] == "Critical")).sum()),
        "Near misses": int((events["event_type"] == "Near miss").sum()),
        "Overdue actions": int((events["overdue_flag"] == True).sum()),
        "High-risk locations": int(events.loc[events["risk_score"] >= 70, "location"].nunique()),
        "Days without LTI": 126,
    }

File: src/utils/safetyCalculations.ts
import { SafetyEvent } from '../types/maritime';

export function safetyKpis(events: SafetyEvent[]): Record<string, string | number> {
  const openEvents = events.filter((e) => e.status !== 'Closed');
  const critical = openEvents.filter((e) => e.severity === 'Critical').length;
  const nearMisses = events.filter((e) => e.event_type === 'Near miss').length;
  const overdue = events.filter((e) => Boolean(e.overdue_flag)).length;

  const highRiskLocations = new Set(
    events.filter((e) => (e.risk_score || 0) >= 70).map((e) => e.location)
  ).size;

  return {
    'Open safety events': openEvents.length,
    'Critical events': critical,
    'Near misses': nearMisses,
    'Overdue actions': overdue,
    'High-risk locations': highRiskLocations,
    'Days without LTI': 126,
  };
}
